phpstan.neon

```diff
@@ -0,0 +1,11 @@
+includes:
+    - ./vendor/larastan/larastan/extension.neon
+
+parameters:
+    paths:
+        - src/
+        - tests/
+        - routes/
+    ignoreErrors:
```

Comment:
I see quite a few PHPStan ignores throughout the code base which I don't really like personally.

* Relations that are referenced using a configurable model are ignored. Wouldn't it be better to just ignore the relations all together? New relations require the ignore as well and it really defeats the point of checking them with static analysis.
* Perhaps the same applies to `Eventy::filter`
* You have added a lot of doc blocks for return types and parameters for arrays and collections. These errors are reported by the `missingType.generics` and `missingType.iterableValue` identifiers. It takes a lot of work to implement and maintain these across Rapidez packages. Do we want to include these rules?

This, of course, is how I think about it. I don't actively work on Rapidez, so I wonder what the other reviews think about this.

Reply:
Fixing the `Eventy::filter` errors is tricky because it returns one generic error alongside a specific error, e.g.:

```
  34     Parameter #3 $mapping of method Rapidez\Core\Commands\ElasticsearchIndexCommand::prepareIndexerWithStore() expects array<string, mixed>, null given.
  34     Result of static method TorMorten\Eventy\Events::filter() (void) is used.
```